core/iwasm/common/wasm_memory.c

```diff
@@ -11,16 +11,25 @@
 typedef enum Memory_Mode {
     MEMORY_MODE_UNKNOWN = 0,
     MEMORY_MODE_POOL,
+    MEMORY_MODE_SYSTEM_ALLOCATOR,
```

Comment:
How about moving `MEMORY_MODE_SYSTEM_ALLOCATOR` after `MEMORY_MODE_ALLOCATOR`, same as order in wasm_export.h?

Reply:
Will also reorder the if/else chains to match